### DISCUSSION TITLE
Inverter status in text instead of numbers

### DISCUSSION THREAD

**Discussion by DorwinDT** [johanmeijer/grott#534] [General]
Hello
I see in diagnostics of my inverter also status. But it is number:
![SmartSelect_20240312_185236_Home Assistant](https://github.com/johanmeijer/grott/assets/17495311/69e89d4d-bbf5-4b57-a274-e8c7a0980ffd)

Is there some list of all statuses? I know only
9 = bypass
5 = on-Grid mode
1= self inspection

Thank you for help

**Comment by johanmeijer** [comment]
You can find status in the growatt protocol documentation (https://github.com/johanmeijer/grott/blob/Beta-(2.8.x)/documentatie/Growatt-Inverter-Modbus-RTU-Protocol-II-V1-24-English-new.pdf).   But is a depending on your inverter type and there are multiple fields describing the status of the inverter. 

To be honest I can not found the status you described here. 

What inverter type do you have?

**Reply by DorwinDT** [reply]
I have Growatt SPH4 10000 TL3 BH-UP
and I know, status=9 is not there. but in reality inverter returns '9' during the night. and it is visible on https://server.growatt.com/ as 'Bypass'
anyway in meantime  I have made node-red function for converting INT state to text:

```
var pv1 = Number(global.get("homeassistant").homeAssistant.states["sensor.XXXXXXXx_sp_state"].state);
switch (pv1)
{
    case 0:
        return { payload: "standby" };
        break;
    case 1:
        return { payload: "self inspection" };
        break;
    case 5:
        return { payload: "On-grid mode" };
        break;
    case 9:
        return { payload: "Bypass" };
        break;        
    default: 
        return { payload: pv1 };
}
```
